src/baseline_rule.py

```diff
@@ -89,33 +91,53 @@ def matches_target(self, labels):
         return BaselineRule._matches_selectors(labels, self.target)
 
     @staticmethod
-    def selectors_as_netpol_peer(selectors):
+    def selectors_as_netpol_peer(selectors, ns_selector=False):
         if not selectors:
+            if ns_selector:
+                # undefined refers to all, in k8s, existing but empty namespaceSelector matches all namespaces
+                return {'namespaceSelector': {}}
             return {}
         if isinstance(selectors, IpSelector):
+            if ns_selector:
+                print('Error: namespaceSelector might not be a CIDR')
+                return None
             return {'ipBlock': selectors.get_cidr()}
         if all(len(selector.values) == 1 and selector.operator == SelectorOp.IN for selector in selectors):
             sel = {'matchLabels': {selector.key: selector.values[0] for selector in selectors}}
         else:
             sel = {'matchExpressions': [selector.convert_to_label_selector_requirement() for selector in selectors]}
+        if ns_selector:
+            return {'namespaceSelector': sel}
         return {'podSelector': sel}
 
     @staticmethod
-    def _selectors_as_netpol_peer_calico(selectors, limit_to_all_expr=True):
+    def _selectors_as_netpol_peer_calico(selectors, limit_to_all_expr=True, ns_selector=False):
         """
         :param Union(list[LabelSelector], IpSelector) selectors: the source or target selectors
         :param bool limit_to_all_expr: should limit calico selector to 'all()' in case of empty input selectors
         :return: calico selector dict
         :rtype dict
         """
         if not selectors:
+            if ns_selector:
+                return {}  # undefined namespaceSelector is equal to {'namespaceSelector': 'all()'}
             return {'selector': 'all()'} if limit_to_all_expr else {}
         if isinstance(selectors, IpSelector):
+            if ns_selector:
+                print('Error: namespaceSelector might not be a CIDR')
+                return None
             return {'nets': selectors.get_nets_calico()}
 
         expr = ' && '.join(selector.convert_to_calico_selector_expression() for selector in selectors)
+        if ns_selector:
+            return {'namespaceSelector': expr}
         return {'selector': expr}
 
+    def _get_namespace_name_from_selector(self, sel):
+        k8s_ns_label = 'kubernetes.io/metadata.name'
+        selector_dict = self.selectors_as_netpol_peer(sel, True)
+        return selector_dict['namespaceSelector']['matchLabels'].get(k8s_ns_label) or None
```

Comment:
You might get a dictionary with just `matchExpression` and not `matchLabels`

Reply:
a question:
lets assume the policy type is Ingress
Should I still build a **k8s networkpolicy** if the target's namespaceSelector contains other labels/expressions than the name of namespace? (i.e. the returned `selector_dict` Is not in the form of matchLabels with only one key: value and `key==kubernetes.io/metadata.name`)
example:

>  target_ns =kubernetes.io/metadata.name=<ns_name>, user = alice 

with current case the metadata will include namespace: <ns_name>
but the other label will be lost!! (Is there a way to add it to the metadata?)

Reply:
Yes, you are right. It's more complicated than I thought.
@adisos , what do you think? should we give up synthesizing baseline rules into K8s NetworkPolicies and only support synthesizing into Calico's GlobalNetworkPolicies? I believe handling namespaces will be much easier this way.

Reply:
Yes, it does sound simpler this way.

Reply:
@shireenf-ibm - Let's remove `BaselineRule.to_netpol()` and all the methods solely used by it.
We'll leave `BaselineRule.to_global_netpol_calico()`. With Calico you can simply add `namespaceSelector` to both `source` and `destination` (if required).

Reply:
Hi,
In my last commit I've removed the `BaselineRule.to_netpol()` method.
but kept following methods, because I found them references in other repository in np-guard:
1. `selectors_as_netpol_peer` (kept the 'master' branch version)
2. `get_port_array`

other methods used by `to_netpol()` were removed, including two methods (`sources_as_netpol_peer and targets_as_netpol_peer`) , which were not used at all (not even before)